src/insights_client/tests/test_metrics.py

```diff
@@ -120,6 +117,14 @@ def test_proxy_settings_without_auth(rhsm_config_file_factory):
     assert proxy_settings == {"https": "http://localhost:3128"}
 
 
+def test_proxy_settings_with_empty_auth(rhsm_config_file_factory):
+    rhsm_config_file = rhsm_config_file_factory(proxy_hostname="localhost", proxy_port=3128, proxy_user="", proxy_password="")
+    rhsm_config = ConfigParser()
+    rhsm_config.read(rhsm_config_file.name)
+    proxy_settings = _proxy_settings(rhsm_config)
+    assert proxy_settings == {"https": "http://localhost:3128"}
+
+
```

Comment:
There are no tests that would verify correct handling of a situation when _proxy_hostname_ key is present, but its value is empty. The first suggested tests verifies a properly empty string, the second one checks the stripping.

```suggestion

def test_proxy_settings_empty_hostname(rhsm_config_file_factory):
    rhsm_config_file = rhsm_config_file_factory(proxy_hostname="")
    rhsm_config = ConfigParser()
    rhsm_config.read(rhsm_config_file.name)
    proxy_settings = _proxy_settings(rhsm_config)
    assert proxy_settings is None


def test_proxy_settings_whitespace_hostname(rhsm_config_file_factory):
    rhsm_config_file = rhsm_config_file_factory(proxy_hostname=" ")
    rhsm_config = ConfigParser()
    rhsm_config.read(rhsm_config_file.name)
    proxy_settings = _proxy_settings(rhsm_config)
    assert proxy_settings is None


```